src/java.security.jgss/share/classes/sun/security/krb5/internal/crypto/EType.java

```diff
@@ -196,50 +237,9 @@ public int startOfPad(byte[] data) {
         return result;
     }
 
-    // Note: the first 2 entries of BUILTIN_ETYPES and BUILTIN_ETYPES_NOAES256
-    // should be kept DES-related. They will be removed when allow_weak_crypto
-    // is set to false.
-
-    private static final int[] BUILTIN_ETYPES = new int[] {
-            EncryptedData.ETYPE_AES256_CTS_HMAC_SHA1_96,
-            EncryptedData.ETYPE_AES128_CTS_HMAC_SHA1_96,
-            EncryptedData.ETYPE_AES256_CTS_HMAC_SHA384_192,
-            EncryptedData.ETYPE_AES128_CTS_HMAC_SHA256_128,
-            EncryptedData.ETYPE_DES3_CBC_HMAC_SHA1_KD,
-            EncryptedData.ETYPE_ARCFOUR_HMAC,
-            EncryptedData.ETYPE_DES_CBC_CRC,
-            EncryptedData.ETYPE_DES_CBC_MD5,
-    };
-
-    private static final int[] BUILTIN_ETYPES_NOAES256 = new int[] {
-            EncryptedData.ETYPE_AES128_CTS_HMAC_SHA1_96,
-            EncryptedData.ETYPE_AES128_CTS_HMAC_SHA256_128,
-            EncryptedData.ETYPE_DES3_CBC_HMAC_SHA1_KD,
-            EncryptedData.ETYPE_ARCFOUR_HMAC,
-            EncryptedData.ETYPE_DES_CBC_CRC,
-            EncryptedData.ETYPE_DES_CBC_MD5,
-    };
-
-
     // used in Config
     public static int[] getBuiltInDefaults() {
-        int allowed = 0;
-        try {
-            allowed = Cipher.getMaxAllowedKeyLength("AES");
-        } catch (Exception e) {
-            // should not happen
-        }
-        int[] result;
-        if (allowed < 256) {
-            result = BUILTIN_ETYPES_NOAES256;
-        } else {
-            result = BUILTIN_ETYPES;
-        }
-        if (!allowWeakCrypto) {
-            // The last 4 etypes are now weak ones
-            return Arrays.copyOfRange(result, 0, result.length - 4);
-        }
-        return result;
+        return defaultETypes;
```

Comment:
It might be safer to return a clone here since it is mutable. The previous code always returned a new array. This array gets passed back to calling code via Etype.getDefaults(), returning a clone would prevent the configured value from being accidentally modified.

Reply:
OK.